src/components/common/PaymentButton.tsx

```diff
@@ -0,0 +1,127 @@
+  const { setVisible: setAuthModalVisible } = useContext(AuthModalContext)
+  const { settings, id: appId } = useApp()
+  const sessionStorageKey = `lodestar.sharing_code.${type}_${target}`
+  const [sharingCode = window.sessionStorage.getItem(sessionStorageKey)] = useQueryParam('sharing', StringParam)
+  sharingCode && window.sessionStorage.setItem(sessionStorageKey, sharingCode)
+  const history = useHistory()
+  const { resourceCollection } = useResourceCollection([`${appId}:${camelCaseToSnake(type)}:${target}`])
+
+  const handleAddCart = () => {
+    return addCartProduct?.(type, target, {
+      quantity,
+      from: window.location.pathname,
+      sharingCode,
+    }).catch(handleError)
+  }
+
+  return (
+    <>
+      {isProductInCart?.(type, target) ? (
+        <Button colorScheme="primary" isFullWidth onClick={() => history.push(`/cart`)}>
+          {formatMessage(commonMessages.button.cart)}
+        </Button>
+      ) : isSubscription ? (
+        <CheckoutProductModal
+          defaultProductId={`${type}_${target}`}
+          renderTrigger={({ isLoading, onOpen }) => (
+            <Button
+              variant="primary"
+              isFullWidth
+              isDisabled={isLoading}
+              onClick={() => {
+                isAuthenticated ? onOpen?.() : setAuthModalVisible?.(true)
+                const resource = resourceCollection.filter(notEmpty)[0]
+                resource && tracking.addToCart(resource, { direct: true })
+                resource && tracking.checkout([resource])
+              }}
+            >
+              {formatMessage(commonMessages.button.subscribeNow)}
+            </Button>
+          )}
+        />
+      ) : currencyId === 'LSC' ? (
+        <CoinCheckoutModal productId={type + '_' + target} currencyId={currencyId} amount={price} />
+      ) : (
+        <div className="d-flex flex-column">
+          {!settings['feature.cart.disable'] && (
+            <StyleButton
+              className="mr-2"
+              variant="outline"
+              colorScheme="primary"
+              isFullWidth
+              isMultiline
+              onClick={() => {
+                resourceCollection[0] && tracking.addToCart(resourceCollection[0])
+                handleAddCart()
+              }}
+            >
+              <Icon as={AiOutlineShoppingCart} />
+              <span className="ml-2">{formatMessage(commonMessages.button.addCart)}</span>
+            </StyleButton>
+          )}
+          <Button
+            colorScheme="primary"
+            isFullWidth
+            onClick={() => {
+              resourceCollection[0] && tracking.addToCart(resourceCollection[0], { direct: true })
+              handleAddCart()?.then(() => history.push('/cart'))
+            }}
+          >
+            {price === 0 ? formatMessage(commonMessages.button.join) : formatMessage(commonMessages.ui.purchase)}
+          </Button>
+        </div>
```

Comment:
1. should be !(setting['feature.cart.disable'] === '1'), beacuse the value would be 0 or 1 (string)

2. it will be display two button , if setting['feature.cart.disable'] pass through condition

Reply:
thanks ~~